hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneClientAdapterImpl.java

```diff
@@ -646,4 +653,56 @@ public String createSnapshot(String pathStr, String snapshotName)
         ofsPath.getBucketName(),
         snapshotName);
   }
+
+  @Override
+  public SnapshotDiffReport getSnapshotDiffReport(Path snapshotDir,
+      String fromSnapshot, String toSnapshot)
+      throws IOException, InterruptedException {
+    boolean takeTemporarySnapshot = false;
+    if (toSnapshot.equals(ACTIVE_FS_SNAPSHOT_NAME)) {
+      takeTemporarySnapshot = true;
+      toSnapshot = createSnapshot(snapshotDir.toString(),
+          "temp" + SnapshotInfo.generateName(Time.now()));
+    }
+    try {
+      SnapshotDiffReportOzone aggregated;
+      SnapshotDiffReportOzone report =
+          getSnapshotDiffReportOnceComplete(fromSnapshot, toSnapshot, "");
+      aggregated = report;
+      while (!report.getToken().isEmpty()) {
+        LOG.info(
+            "Total Snapshot Diff length between snapshot {} and {} exceeds"
+                + " max page size, Performing another " +
+                "snapdiff with index at {}",
+            fromSnapshot, toSnapshot, report.getToken());
+        report = getSnapshotDiffReportOnceComplete(fromSnapshot, toSnapshot,
+            report.getToken());
+        aggregated.aggregate(report);
+      }
+      return aggregated;
+    } finally {
+      // delete the temp snapshot
+      if (takeTemporarySnapshot) {
+        OFSPath snapPath = new OFSPath(snapshotDir.toString(), config);
+        ozoneClient.getObjectStore()
+            .deleteSnapshot(snapPath.getVolumeName(), snapPath.getBucketName(),
```

Comment:
Yes thats a good point, there is a snapshot deleting task that cares care of cleaning up the deleted snapshots, we could add handling there to delete these type of snapshots if present.
cc @smengcl

Reply:
Temp snapshots should not be part of Garbage collection on their own. We need to have a separate service to  issue deletes on the temporary snapshots. This can be a separate Jira and a separte PR.